subcommand/tls-init/commant_test.go

```diff
@@ -0,0 +1,582 @@
+package tls_init
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestRun_FlagValidation(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		flags  []string
+		expErr string
+	}{
+		{
+			flags:  nil,
+			expErr: "-name-prefix must be set",
+		},
+		{
+			flags:  []string{"-name-prefix", "consul", "-ca", "foo"},
+			expErr: "either both -ca and -key or neither must be set",
+		},
+		{
+			flags:  []string{"-name-prefix", "consul", "-key", "/foo"},
+			expErr: "either both -ca and -key or neither must be set",
+		},
+		{
+			flags:  []string{"-name-prefix", "consul", "-log-level", "invalid"},
+			expErr: "unknown log level: invalid",
+		},
+		{
+			flags:  []string{"-name-prefix", "consul", "-days", "-3"},
+			expErr: "-days must be a positive integer",
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.expErr, func(tt *testing.T) {
+			ui := cli.NewMockUi()
+			cmd := Command{UI: ui}
+			exitCode := cmd.Run(c.flags)
+			require.Equal(tt, 1, exitCode, ui.ErrorWriter.String())
+			require.Contains(tt, ui.ErrorWriter.String(), c.expErr)
+		})
+	}
+}
+
+func TestRun_CreatesServerCertificatesWithExistingCertsAsFiles(t *testing.T) {
+	ui := cli.NewMockUi()
+	cmd := Command{UI: ui}
+	k8s := fake.NewSimpleClientset()
+	cmd.clientset = k8s
+
+	ca, err := ioutil.TempFile("", "")
+	require.NoError(t, err)
+	err = ioutil.WriteFile(ca.Name(), []byte(caCert), 0644)
+	require.NoError(t, err)
```

Comment:
need to add a `defer` or a `t.Cleanup` to remove files.

Reply:
Good catch. Thanks 😺

Reply:
done :)